openwisp_notifications/handlers.py

```diff
@@ -192,34 +223,32 @@ def send_email_notification(sender, instance, created, **kwargs):
     if not (email_preference and instance.recipient.email and email_verified):
         return
 
-    try:
-        subject = instance.email_subject
-    except NotificationRenderException:
-        # Do not send email if notification is malformed.
+    recipient_id = instance.recipient.id
+    cache_key = f'email_batch_{recipient_id}'
+    cache_key_pks = f'{instance.recipient.email}_pks'
+    cache_data = cache.get(
+        cache_key, {'last_email_sent_time': None, 'batch_scheduled': False}
+    )
+
+    if cache_data['last_email_sent_time'] and EMAIL_BATCH_INTERVAL > 0:
+        if not cache_data['batch_scheduled']:
+            # Schedule batch email notification task
+            tasks.batch_email_notification.apply_async(
+                (instance.recipient.email,), countdown=EMAIL_BATCH_INTERVAL
+            )
+            cache_data['batch_scheduled'] = True
+            cache.set(cache_key_pks, [instance.id], timeout=EMAIL_BATCH_INTERVAL * 2)
```

Comment:
We should not set a timeout for the cache here. It may be possible that the celery worker is overloaded. It could take more time than the set timeout for execution of the batch email sending task. We should only clear the cache key once the batch email processing is complete.